pkg/cri/resource-manager/requests.go

```diff
@@ -461,16 +466,77 @@ func (m *resmgr) UpdateContainer(ctx context.Context, method string, request int
 	return &criapi.UpdateContainerResourcesResponse{}, nil
 }
 
+type ContainerStats struct {
+	CPU    float64 `containermetrics:"cpu"`
+	Memory float64 `containermetrics:"wss"`
+}
+
+type NumaMetrics struct {
+	Total       float64 `containermetrics:"numa_total"`
+	File        float64 `containermetrics:"numa_file"`
+	Anon        float64 `containermetrics:"numa_anon"`
+	Unevictable float64 `containermetrics:"numa_unevictable"`
+}
+
 // ListContainerStats feeds CRI container statistics to metrics processing.
 func (m *resmgr) ListContainerStats(ctx context.Context, method string, request interface{},
 	handler server.Handler) (interface{}, error) {
 
 	reply, rqerr := handler(ctx, request)
 
+	numaStats := make(map[string]cgroups.NumaStat)
 	filter := request.(*criapi.ListContainerStatsRequest).Filter
 	if filter.Id == "" && filter.PodSandboxId == "" && len(filter.LabelSelector) == 0 {
-		if err := m.SendEvent(reply.(*criapi.ListContainerStatsResponse).Stats); err != nil {
-			m.Error("failed to inject CRI ListContainerStats for event processing: %v", err)
```

Comment:
The reason for the original code to send practically the raw message to the event loop for processing is that we only intercept this request to use its reply internally, without any intention to modify either. Therefore it is possible to minimize the additional delay introduced by cri-resmgr being on the signalling path, if we do the local processing in the event loop which is running in a separate goroutine. So I thin all this processing should be moved to the other end of the event channel.

Reply:
moved all processing in the handler